e2e/utils/test-utils.ts

```diff
@@ -254,57 +272,55 @@ export async function createWorkspace(
  *  workspaceName - return the workspace with this name if it can be found; default behavior otherwise
  */
-export async function findOrCreateWorkspace(
-  page: Page,
-  opts: { alwaysCreate?: boolean; workspaceName?: string } = {}
-): Promise<WorkspaceCard> {
-  const { alwaysCreate = false, workspaceName } = opts;
-
-  const workspacesPage = new WorkspacesPage(page);
-  await workspacesPage.load();
+export async function findOrCreateWorkspace(page: Page, opts: { workspaceName?: string } = {}): Promise<WorkspaceCard> {
+  const { workspaceName } = opts;
 
-  const workspaceCard = new WorkspaceCard(page);
   // Returns specified workspaceName Workspace card if exists.
   if (workspaceName !== undefined) {
-    const cardFound = await workspaceCard.findCard(workspaceName);
+    const cardFound = await findWorkspaceCard(page, workspaceName);
     if (cardFound != null) {
-      // Workspace card found
-      return cardFound;
+      return cardFound; // Found Workspace card matching workspace name
     }
   }
 
   // Find all workspaces with OWNER role
+  const workspacesPage = new WorkspacesPage(page);
+  await workspacesPage.load();
   const existingCards = await WorkspaceCard.findAllCards(page, WorkspaceAccessLevel.Owner);
-
   // Filter to include Workspaces older than 30 minutes
-  const halfHourMillisec = 1000 * 60 * 30;
+  const halfHourMilliSec = 1000 * 60 * 30;
   const now = Date.now();
   const olderWorkspaceCards = [];
   for (const card of existingCards) {
     const workspaceTime = Date.parse(await card.getLastChangedTime());
     const timeDiff = now - workspaceTime;
-    if (timeDiff > halfHourMillisec) {
+    if (timeDiff > halfHourMilliSec) {
       olderWorkspaceCards.push(card);
     }
   }
 
-  // Create new workspace if existing workspace is zero or alwayCreate is true
-  if (alwaysCreate || olderWorkspaceCards.length === 0) {
+  // Create new workspace if existing workspace is zero or alwaysCreate is true
+  if (olderWorkspaceCards.length === 0 || workspaceName !== undefined) {
     const name = workspaceName || makeWorkspaceName();
-    await workspacesPage.createWorkspace(name);
-    console.log(`Created workspace "${name}"`);
-    await workspacesPage.load();
-    return workspaceCard.findCard(name);
+    const [card] = await createWorkspace(page, { workspaceName: name, openDataPage: false });
+    return card;
   }
 
   // Return one random Workspace card
   const randomCard = fp.shuffle(olderWorkspaceCards).pop();
   const cardName = await randomCard.getWorkspaceName();
   const lastChangedTime = await randomCard.getLastChangedTime();
-  console.log(`Found workspace "${cardName}". Last changed on ${lastChangedTime}`);
+  console.log(`Found workspace card: "${cardName}". Last changed on ${lastChangedTime}`);
   return randomCard;
 }
 
+export async function findWorkspaceCard(page: Page, name: string): Promise<WorkspaceCard | null> {
```

Comment:
Extracted out common code - find-workspace-card from `createWorkspace` and `findOrCreateWorkspace` functions.